Bottom: 27f4cfec5787900aebe1d95965e795804ea10d94
Top:    b582a06f20c72984bb0225c5507036502940e64c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-14 14:39:04 +0100

Refresh of drop-in-pcb-gl-code-various-me

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 739f29e..0c66353 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -831,9 +831,9 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
   glLoadIdentity ();
   glTranslatef (0.0f, 0.0f, -Z_NEAR);
 
-  glClearColor (gport->bg_color.red / 65535.,
-                gport->bg_color.green / 65535.,
-                gport->bg_color.blue / 65535.,
+  glClearColor (gport->offlimits_color.red / 65535.,
+                gport->offlimits_color.green / 65535.,
+                gport->offlimits_color.blue / 65535.,
                 1.);
 
   glClear (GL_COLOR_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
@@ -843,60 +843,18 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
   region.Y1 = MIN (Py (ev->area.y), Py (ev->area.y + ev->area.height + 1));
   region.Y2 = MAX (Py (ev->area.y), Py (ev->area.y + ev->area.height + 1));
 
-  eleft = Vx (0);
-  eright = Vx (PCB->MaxWidth);
-  etop = Vy (0);
-  ebottom = Vy (PCB->MaxHeight);
-  if (eleft > eright)
-    {
-      int tmp = eleft;
-      eleft = eright;
-      eright = tmp;
-    }
-  if (etop > ebottom)
-    {
-      int tmp = etop;
-      etop = ebottom;
-      ebottom = tmp;
-    }
+  eleft = Vx (0);  eright  = Vx (PCB->MaxWidth);
+  etop  = Vy (0);  ebottom = Vy (PCB->MaxHeight);
 
-  glColor3f (gport->offlimits_color.red / 65535.,
-             gport->offlimits_color.green / 65535.,
-             gport->offlimits_color.blue / 65535.);
+  glColor3f (gport->bg_color.red / 65535.,
+             gport->bg_color.green / 65535.,
+             gport->bg_color.blue / 65535.);
 
   glBegin (GL_QUADS);
-  if (eleft > 0)
-    {
-      glVertex2i (0, 0);
-      glVertex2i (eleft, 0);
-      glVertex2i (eleft, gport->height);
-      glVertex2i (0, gport->height);
-    }
-  else
-    eleft = 0;
-  if (eright < gport->width)
-    {
-      glVertex2i (eright, 0);
-      glVertex2i (gport->width, 0);
-      glVertex2i (gport->width, gport->height);
-      glVertex2i (eright, gport->height);
-    }
-  else
-    eright = gport->width;
-  if (etop > 0)
-    {
-      glVertex2i (eleft, 0);
-      glVertex2i (eright, 0);
-      glVertex2i (eright, etop);
-      glVertex2i (eleft, etop);
-    }
-  if (ebottom < gport->height)
-    {
-      glVertex2i (eleft, ebottom);
-      glVertex2i (eright + 1, ebottom);
-      glVertex2i (eright + 1, gport->height);
-      glVertex2i (eleft, gport->height);
-    }
+  glVertex3i (eleft,  etop,    0);
+  glVertex3i (eright, etop,    0);
+  glVertex3i (eright, ebottom, 0);
+  glVertex3i (eleft,  ebottom, 0);
   glEnd ();
 
   /* TODO: Background image */
